client/src/app/shared/authentication/require-unauthenticated-guard.ts

```diff
@@ -0,0 +1,46 @@
+import { Injectable } from '@angular/core';
+import {
+  ActivatedRouteSnapshot,
+  CanActivate,
+  CanLoad,
+  Route,
+  RouterStateSnapshot,
+  UrlSegment,
+  UrlTree,
+} from '@angular/router';
+import { Observable } from 'rxjs';
+import { map } from 'rxjs/operators';
+
+import { AuthService } from '@auth0/auth0-angular';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class RequireUnauthenticatedGuard implements CanActivate, CanLoad {
+  constructor(private authService: AuthService) {}
+  canActivate(
+    route: ActivatedRouteSnapshot,
+    state: RouterStateSnapshot,
+  ):
+    | boolean
+    | UrlTree
+    | Observable<boolean | UrlTree>
+    | Promise<boolean | UrlTree> {
+    return this.authService.isAuthenticated$.pipe(
+      map((b) => !b),
+    );
+  }
+
+  canLoad(
+    route: Route,
+    segments: UrlSegment[],
+  ):
+    | boolean
+    | UrlTree
+    | Observable<boolean | UrlTree>
+    | Promise<boolean | UrlTree> {
+    return this.authService.isAuthenticated$.pipe(
+      map((b) => !b),
```

Comment:
I think, generally, you don't want to allow loading of a module (downloading it w/ async routes), if a user won't be able to activate it.

Was there an issue with this method having the same logic as `canActivate`?